xonsh/environ.py

```diff
@@ -176,7 +176,7 @@ def DEFAULT_ENSURERS():
     'XONSH_DEBUG': (always_false, to_debug, bool_or_int_to_str),
     'XONSH_ENCODING': (is_string, ensure_string, ensure_string),
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
-    'XONSH_HISTORY_BACKEND': (is_string, ensure_string, ensure_string),
+    'XONSH_HISTORY_BACKEND': (is_string_or_callable, ensure_string, ensure_string),
```

Comment:
`ensure_string` is probably no longer the right thing here. It is probably worth writing a custom to-string converter here that reads the history classes `.name` attribute. Of course, we'd need to add the name attr to the existing backends.